src/video/wayland/SDL_waylandwindow.c

```diff
@@ -894,6 +899,15 @@ Wayland_MaximizeWindow(_THIS, SDL_Window * window)
     SDL_WindowData *wind = window->driverdata;
     SDL_VideoData *viddata = (SDL_VideoData *) _this->driverdata;
 
+    if (!(window->flags & SDL_WINDOW_RESIZABLE)) {
+        return;
+    }
```

Comment:
I think `Wayland_MaximizeWindow` should not return early if `SDL_WINDOW_RESIZABLE` is not set.

As the application (that uses SDL) will decide when to resize a window and it will also be aware of the window state, it should stay responsible of making these decisions. SDL should merely be a wrapper around these basic functions and the functions should have according to their names. `MaximizeWindow` should forward this request to the window manager and not silently inhibit this request from the application.

Reply:
Unfortunately the toplevel function does not behave correctly without the flag - it treats maximize behavior as separate from window size, so the end result is a window that's pinned but still fixed-size, which is not what we want.

The easiest test is testwm2, x11 and wayland need to behave the same.